sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_appinsights_utils.py

```diff
@@ -0,0 +1,128 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import logging
+
+from azure.ai.ml._azure_environments import _get_base_url_from_metadata
+from azure.ai.ml._vendor.azure_resources._resource_management_client import ResourceManagementClient
+from azure.ai.ml.constants._common import ArmConstants
+from azure.core.credentials import TokenCredential
+from azure.core.exceptions import (HttpResponseError)
+
+module_logger = logging.getLogger(__name__)
+
+
+def default_resource_group_for_app_insights_exists(
+    credentials: TokenCredential,
+    subscription_id: str,
+    location: str) -> bool:
+    client = ResourceManagementClient(
+        credential=credentials,
+        subscription_id=subscription_id,
+        base_url=_get_base_url_from_metadata(),
+        api_version=ArmConstants.AZURE_MGMT_RESOURCE_API_VERSION,
+    )
+    default_rgName = get_default_resource_group_name(location)
+    try:
+        client.resource_groups.get(resource_group_name=default_rgName)
+        return True
+    except HttpResponseError:
+        return False
+
+
+def default_log_analytics_workspace_exists(credentials: TokenCredential, subscription_id: str, location: str) -> bool:
+    client = ResourceManagementClient(
+        credential=credentials,
+        subscription_id=subscription_id,
+        base_url=_get_base_url_from_metadata(),
+        api_version=ArmConstants.AZURE_MGMT_RESOURCE_API_VERSION,
+    )
+    default_resource_group = get_default_resource_group_name(location)
+    default_workspace = client.resources.list_by_resource_group(
+        default_resource_group,
+        filter="substringof('%s',name)"%get_default_log_analytics_name(location)
```

Comment:
```suggestion
        filter=f"substringof('{get_default_log_analytics_name(location)}',name)"
```

prefer f-string over % interpolation